pkg/registry/event/strategy.go

```diff
@@ -59,6 +60,9 @@ func (eventStrategy) AllowCreateOnUpdate() bool {
 
 func (eventStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object) fielderrors.ValidationErrorList {
 	event := obj.(*api.Event)
+	if event.ObjectMeta.UID == "" {
+		event.ObjectMeta.UID = util.NewUUID()
```

Comment:
+1. This is mainly for events that get updated and so we need to preserve the UID.